server/auth/auth.go

```diff
@@ -153,22 +153,29 @@ func (client *clientInfo) pruneBonds(lockTimeThresh int64) (pruned []*db.Bond, b
 		return
 	}
 
-	var n int
+	pruned = make([]*db.Bond, 0, len(client.bonds))
+	expiredBondsInfo := make([]string, 0) // for logging
+	activeCount := 0
+
 	for _, bond := range client.bonds {
-		if bond.LockTime >= lockTimeThresh { // not expired
-			if len(pruned) > 0 /* n < i */ { // a prior bond was removed, must move this element up in the slice
-				client.bonds[n] = bond
-			}
-			n++
-			bondTier += int64(bond.Strength)
+		if bond.LockTime < lockTimeThresh { // bond is expired
+			expiredBondsInfo = append(expiredBondsInfo, fmt.Sprintf("Expiring user %v bond %v (%s)", client.acct.ID, coinIDString(bond.AssetID, bond.CoinID), dex.BipIDSymbol(bond.AssetID)))
```

Comment:
Let's undo this change and just log it here. This doesn't provide any obvious benefit and new allocations and a second loop.